File: Data Structure&Algorithm/meeting/wangyi/demo3.py
def main():
    m=int(input())
    while m:
        n,k=list(map(int,input().split(" ")))
        ans=0
        if k>n-k:
            ans=(n-k)
        else:
            temp=max(k-1,0)
            ans=min(n-k,temp)
        print(0,ans)
        m-=1
main()

File: Data Structure&Algorithm/meeting/tengxun/before/demo3.cpp
#include<iostream>
using namespace std;
int main()
{
    int n,m;
    cin>>n>>m;
    int temp=n/(m*2);
    int rem=n%(m*2);
    int ans=0;
    if(rem>0){
        for(int i=n;i>n-m;i--)
           ans+=i;
    }
    ans=temp*m*m-ans;
    cout<<ans<<endl;
}
// -1 -2 -3 4 5 6 -7 -8 -9